tests/json_infra/conftest.py

```diff
@@ -125,6 +131,36 @@ def pytest_collection_modifyitems(config: Config, items: list[Item]) -> None:
         items[:] = selected  # keep only what matches
 
 
+def _tests_path_collection_modifyitems(
+    config: Config, items: list[Item]
+) -> None:
+    tests_path = config.getoption("tests_path", None)
+    if tests_path is None:
+        return
+
+    with open(tests_path) as f:
+        test_ids = set(x[:-1] for x in f.readlines())
+
+    selected = []
+    deselected = []
+    for item in items:
+        if item.nodeid in test_ids:
+            selected.append(item)
+            test_ids.remove(item.nodeid)
+        else:
+            deselected.append(item)
+
+    if deselected:
+        config.hook.pytest_deselected(items=deselected)
+        items[:] = selected  # keep only what matches
+
+
+def pytest_collection_modifyitems(config: Config, items: list[Item]) -> None:
```

Comment:
I think this hook will no longer be used if we go ahead with the changes proposed in #1666